src/content/en/updates/2017/12/device-memory.md

```diff
@@ -0,0 +1,265 @@
+  include the specified client hints headers in all subsequent request from
+  this origin for the number of seconds specified by the header value.
+</aside>
+
+### Using the JavaScript API
+
+In some cases (like with a static file server or a CDN) you won't be able to
+conditionally respond to requests based on an HTTP header. In these cases you
+can use the JavaScript API to make conditional requests in your JavaScript code.
+
+The following logic is very similar to the Express route above, except it
+dynamically determines the script URL in the client-side logic:
+
+
+```
+// Low-memory devices should load the "lite" version of the component.
+// The logic below will set `componentVersion` to "lite" if (and only if)
+// deviceMemory isn't undefined and returns a number less than 1.
+const componentVersion = navigator.deviceMemory < 1 ? 'lite' : 'full';
+
+const component = await import(`path/to/component.${componentVersion}.js`);
+component.init();
+```
+
+While conditionally serving different versions of the same component based on
+device capabilities is a good strategy, sometimes it can be even better to not
+serve a component at all.
+
+In many cases, components are purely enhancements. They add some nice touches to
+the experience, but they aren't required for the app's core functionality. In
+these cases, it may be wise to not load such components in the first place.
+After all, if a component whose purpose is to improve the user experience
+actually ends up making the app load slower or less responsive to user input, is
+it really achieving its goal?
+
+With any decision you make that affects the user experience, it's critical that
+you're measuring its impact. It's also critical that you have a clear picture of
+how your app performs today.
+
+Understanding how device memory correlates with user behavior for the current
+version of your app will better inform what action needs to be taken, and it'll
+give you a baseline against which you can measure the success of future changes.
+
+
+## Tracking device memory with analytics
+
+The Device Memory API is very new, and most analytics providers are not tracking
+it by default. However, most analytics providers give you a way to track custom
+data (for example, Google Analytics has a feature called [Custom
+Dimensions](https://support.google.com/analytics/answer/2709828)), and you can
+use that to track Device Memory for you users devices.
+
+### Using a custom Device Memory dimension
+
+Using custom dimensions in Google Analytics is a two-step process.
+
+1. First you have to [set up the custom dimension](
+   https://support.google.com/analytics/answer/2709829) in Google Analytics
+2. Next you have to update your tracking code to [`set`](
+   /analytics/devguides/collection/analyticsjs/accessing-trackers)
+   the device memory value for the custom dimension you just created.
+
+When creating the custom dimension, give it the name "Device Memory" and choose
```

Comment:
In this case it's the name of the custom dimension, and I'm capitalizing it since that's how other dimensions and metrics are formatted in Google Analytics.